test/test_log.py

```diff
@@ -156,6 +157,15 @@ def test_long_string_logging(self, backend: FakeModelBackend, logger: logging.Lo
         assert len(calls[1][1]) == MAX_LOG_LINE_LEN
         assert len(calls[2][1]) == 9
 
+    def test_warning(self, backend: FakeModelBackend):
+        ops.log.setup_root_logging(backend)
+        warnings.warn('warning')
+        calls = backend.calls()
+        assert len(calls) == 1
+        assert calls[0][0] == 'WARNING'
```

Comment:
Shouldn't we test that the correct backend method is called, specifically `juju_log`? (Or use `FakeScript` and fake out the `juju-log` script itself?)

Reply:
In this test file, the whole backend is mocked, rather than just the `juju_log` method. And `FakeModelBackend.juju_log` append items to `backend.calls()`, so it's not necessary to assert that `juju_log` is called, right?